exporter/jaegerexporter/exporter_test.go

```diff
@@ -264,7 +278,10 @@ func TestConnectionStateChange(t *testing.T) {
 		wg.Done()
 	})
 
-	require.NoError(t, sender.start(context.Background(), componenttest.NewNopHost()))
+	go func() {
+		sender.startConnectionStatusReporter()
```

Comment:
This looks odd. Shouldn't you be calling `start` instead?

Reply:
If not, you'd need a way to stop the connection reporter. Otherwise, it leaks resources between tests.

Reply:
Yeah, we need to call `startConnectionStatusReporter` explicitly, calling start will create actual grpc connection which we can't control the states. The shutdown in the next line calls the appropriate termination logic in startConnectionStatusReporter. I have added a signal to ensure we wait for the `startConnectionStatusReporter` comes to end,